### PR TITLE
Add 3.x SignalR background service sample

### LINKED ISSUE #15042: SignalR with background service in Core 3.0
Is there a possibility to have [this](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/signalr/background-service/sample) sample updated for Core 3 (along with the [docs](https://docs.microsoft.com/en-us/aspnet/core/signalr/background-services?view=aspnetcore-3.0) that reference it)? With the new 'Worker Service' template, WebHost has been replaced by Host and I had to install Microsoft.AspNetCore.Signalr NuGet to be able to add SignalR to the service collection, but I could not figure out how to wire up Hub endpoints, as I could not find any ApplicationBuilder, to add UseEndpoints middleware.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1bbb2d1c-4dfe-3faf-d4b0-65ae1146961e
* Version Independent ID: 6b203f18-adfa-3a18-183d-b124325064c3
* Content: [Host ASP.NET Core SignalR in background services](https://docs.microsoft.com/en-us/aspnet/core/signalr/background-services?view=aspnetcore-3.0)
* Content Source: [aspnetcore/signalr/background-services.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/signalr/background-services.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-signalr**
* GitHub Login: @bradygaster
* Microsoft Alias: **bradyg**

### PR DESCRIPTION
**Pull request by UncleDave**
Fixes #15042

I've probably been overly aggressive with the addition of monikers, but now that there are 2 versions of the sample I didn't want to hardcode files that don't change to link to the 2.2 version.

@Rick-Anderson  `edit:`
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/background-services?view=aspnetcore-3.1&branch=pr-en-us-17318)

### CODE REVIEW

File: aspnetcore/signalr/background-service/samples/3.0/Clients.ConsoleTwo/ClockHubClient.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+using HubServiceInterfaces;
+using Microsoft.AspNetCore.SignalR.Client;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Clients.ConsoleTwo
+{
+    public class ClockHubClient : IClock, IHostedService
+    {
+#region ClockHubClientCtor
```
Comment (davidfowl) [inline aspnetcore/signalr/background-service/samples/3.0/Clients.ConsoleTwo/ClockHubClient.cs]: No regions

Comment (davidfowl) [reply aspnetcore/signalr/background-service/samples/3.0/Clients.ConsoleTwo/ClockHubClient.cs]: Oh wait is this for the docs?

Comment (UncleDave) [reply aspnetcore/signalr/background-service/samples/3.0/Clients.ConsoleTwo/ClockHubClient.cs]: The regions are no different to the original sample, do you want them removed? They're linked to in the related docs page.

Comment (UncleDave) [reply aspnetcore/signalr/background-service/samples/3.0/Clients.ConsoleTwo/ClockHubClient.cs]: > Oh wait is this for the docs?

Yes :D

Comment (Rick-Anderson) [reply aspnetcore/signalr/background-service/samples/3.0/Clients.ConsoleTwo/ClockHubClient.cs]: Regions are needed to import snippets. This is a vast improvement to the previous version that pasted snippets into the MD file.

The preferred approach when more than 3 or 4 pair of monikers are used is  use monikers on continuous areas only:

```
::: moniker range=">= aspnetcore-3.0"

// Entire 3.1 version here

::: moniker-end
::: moniker range="< aspnetcore-3.0"

// Entire 2.1 version here

::: moniker-end
```

That makes it much easier to add the 5.0 and 5.1 versions. It's much easier to maintain the doc.

File: aspnetcore/signalr/background-services.md [hunk 2]
```diff
@@ -68,11 +53,29 @@ For more information on strongly typed Hubs, see [Use hubs in SignalR for ASP.NE
 > [!NOTE]
 > This functionality isn't limited to the [Hub\<T>](xref:Microsoft.AspNetCore.SignalR.Hub`1) class. Any class that inherits from [Hub](xref:Microsoft.AspNetCore.SignalR.Hub), such as [DynamicHub](xref:Microsoft.AspNetCore.SignalR.DynamicHub), will also work.
```
Comment (Rick-Anderson) [inline aspnetcore/signalr/background-services.md]:
```suggestion
> This functionality isn't limited to the [Hub\<T>](xref:Microsoft.AspNetCore.SignalR.Hub`1) class. Any class that inherits from [Hub](xref:Microsoft.AspNetCore.SignalR.Hub), such as [DynamicHub](xref:Microsoft.AspNetCore.SignalR.DynamicHub) works.
```

Comment (UncleDave) [reply aspnetcore/signalr/background-services.md]: After looking at this a bit more I think it reads better if the comma and the future tense are kept, but I agree that the "also" is not needed. Happy to defer on this if you disagree, but I'm going to lay the 3 out side by side for easy comparison as it's tricky to really feel out the sentence when there's markdown everywhere.

>Any class that inherits from Hub, such as DynamicHub, will also work.

>Any class that inherits from Hub, such as DynamicHub works.

>Any class that inherits from Hub, such as DynamicHub, will work.

Comment (Rick-Anderson) [reply aspnetcore/signalr/background-services.md]: @UncleDave  the MS style guide specifically recommends avoiding future tense. To the untrained ear, FT often sounds better. It doesn't machine translate and it's actually no clearer.

Comment (UncleDave) [reply aspnetcore/signalr/background-services.md]: Fair enough, how do we feel about just restoring the bracketing comma then?

>Any class that inherits from Hub, such as DynamicHub, works.

Comment (Rick-Anderson) [reply aspnetcore/signalr/background-services.md]: That's fine

File: aspnetcore/signalr/background-service/samples/3.x/Clients.ConsoleTwo/ClockHubClient.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        public ClockHubClient(ILogger<ClockHubClient> logger)
+        {
+            _logger = logger;
+            
+            _connection = new HubConnectionBuilder()
+                .WithUrl(Strings.HubUrl)
+                .Build();
+
+            _connection.On<DateTime>(Strings.Events.TimeSent, 
+                dateTime => _ = ShowTime(dateTime));
```
Comment (BrennanConroy) [inline aspnetcore/signalr/background-service/samples/3.x/Clients.ConsoleTwo/ClockHubClient.cs]: `On` handlers should support async methods.
```suggestion
                dateTime => ShowTime(dateTime));
```

Comment (UncleDave) [reply aspnetcore/signalr/background-service/samples/3.x/Clients.ConsoleTwo/ClockHubClient.cs]: Resolved in b552e67

**Pull request merged**